scene/gui/button.cpp

```diff
@@ -379,7 +388,7 @@ Size2 Button::get_minimum_size_for_text_and_icon(const String &p_text, Ref<Textu
 		minsize.width = 0;
 	}
 
-	if (!expand_icon && p_icon.is_valid()) {
+	if (!expand_icon && icon_set_height == 0 && p_icon.is_valid()) {
```

Comment:
Same here:
```suggestion
	if (!expand_icon && icon_set_height <= 0 && p_icon.is_valid()) {
```

Reply:
Will do - same question as above regarding workflow.

A code question: in the `Button` class, a number of the setters have `update_minimum_size` _after_ `queue_redraw`, but it seems like the size update should always occur first? It occurs first in `_notification`.

Also, is there a trick to disallow negative values in the first place without doing a check and substitution? Is the philosopy to allow them but ignore them when meaningless?